### PR TITLE
New Alsid for AD connector, workbooks and analytics templates

### PR DESCRIPTION
**Pull request by Azomasiel**
# Proposed Changes

  - Alsid for AD Syslog connector
  - Alsid for AD Indicators of exposure workbook
  - Alsid for AD Indicators of exposure analytic rules

### CODE REVIEW

File: Parsers/Alsid/afad_parser.kql [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// Usage Instruction :
```
Comment (shainw) [inline Parsers/Alsid/afad_parser.kql:1]: Parser should follow the standard format, example from this - https://github.com/Azure/Azure-Sentinel/blob/master/Parsers/PulseConnectSecure/PulseConnectSecure.txt.  Need to get a Wiki update to include required template for parsers, but you can use the PulseConnectSecure version as a template.

**Review by shainw — CHANGES_REQUESTED** [review]
Validations will fail on all the Detection YAML's because the table schema cannot be identified.  In order to work around this, you will need to provide a Custom Table schema here - https://github.com/Azure/Azure-Sentinel/tree/master/.script/tests/KqlvalidationsTests/CustomTables.  They are simple JSON format.  The name of the table goes at that top in the name field, in your case it would be "afad_parser".  Then the schema follows in the Properties key value pair section.

File: Detections/AlsidForAD/ADAttacksPathways.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  - T1110
+query: |
+  let SeverityTable=datatable(Severity:string,Level:int) [
+  "low", 1,
+  "medium", 2,
+  "high", 3,
+  "critical", 4
+  ];
+  afad_parser
+  | where MessageType == 0 and (Codename == "C-PRIV-ACCOUNTS-SPN" or Codename == "C-SDPROP-CONSISTENCY" or Codename == "C-DANG-PRIMGROUPID" or Codename == "C-GPO-HARDENING" or Codename == "C-DC-ACCESS-CONSISTENCY" or Codename == "C-DANGEROUS-TRUST-RELATIONSHIP" or Codename == "C-UNCONST-DELEG" or Codename == "C-ABNORMAL-ENTRIES-IN-SCHEMA")
```
Comment (shainw) [inline Detections/AlsidForAD/ADAttacksPathways.yaml:26]: Missed this on the first pass.  Much more readable if you create a list for this and use the 'in' operator.  Change to -

let codeNameList = ("C-PRIV-ACCOUNTS-SPN", "C-SDPROP-CONSISTENCY", "C-DANG-PRIMGROUPID", "C-GPO-HARDENING", "C-DC-ACCESS-CONSISTENCY", "C-DANGEROUS-TRUST-RELATIONSHIP", "C-UNCONST-DELEG", "C-ABNORMAL-ENTRIES-IN-SCHEMA");
afad_parser
| where MessageType == 0 and codeNameList in~ (codeNameList)

File: Detections/AlsidForAD/PasswordIssues.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  - T1110
+query: |
+  let SeverityTable=datatable(Severity:string,Level:int) [
+  "low", 1,
+  "medium", 2,
+  "high", 3,
+  "critical", 4
+  ];
+  afad_parser
+  | where MessageType == 0 and (Codename == "C-CLEARTEXT-PASSWORD" or Codename == "C-PASSWORD-DONT-EXPIRE" or Codename == "C-USER-REVER-PWDS" or Codename == "C-PASSWORD-POLICY" or Codename == "C-USER-PASSWORD" or Codename == "C-KRBTGT-PASSWORD" or Codename == "C-AAD-SSO-PASSWORD" or Codename == "C-REVER-PWD-GPO")
```
Comment (shainw) [inline Detections/AlsidForAD/PasswordIssues.yaml:26]: Change to List, same as I had shown in "Alsid Active Directory attacks pathways" detection.

File: Detections/AlsidForAD/PrivilegedAccountIssues.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  - T1110
+query: |
+  let SeverityTable=datatable(Severity:string,Level:int) [
+  "low", 1,
+  "medium", 2,
+  "high", 3,
+  "critical", 4
+  ];
+  afad_parser
+  | where MessageType == 0 and (Codename == "C-PRIV-ACCOUNTS-SPN" or Codename == "C-NATIVE-ADM-GROUP-MEMBERS" or Codename == "C-KRBTGT-PASSWORD" or Codename == "C-PROTECTED-USERS-GROUP-UNUSED" or Codename == "C-ADMINCOUNT-ACCOUNT-PROPS" or Codename == "C-ADM-ACC-USAGE" or Codename == "C-LAPS-UNSECURE-CONFIG" or Codename == "C-DISABLED-ACCOUNTS-PRIV-GROUPS")
```
Comment (shainw) [inline Detections/AlsidForAD/PrivilegedAccountIssues.yaml:26]: Change to List, same as I had shown in "Alsid Active Directory attacks pathways" detection.

File: Detections/AlsidForAD/UserAccountIssues.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  - T1110
+query: |
+  let SeverityTable=datatable(Severity:string,Level:int) [
+  "low", 1,
+  "medium", 2,
+  "high", 3,
+  "critical", 4
+  ];
+  afad_parser
+  | where MessageType == 0 and (Codename == "C-ACCOUNTS-DANG-SID-HISTORY" or Codename == "C-PRE-WIN2000-ACCESS-MEMBERS" or Codename == "C-PASSWORD-DONT-EXPIRE" or Codename == "C-SLEEPING-ACCOUNTS" or Codename == "C-DANG-PRIMGROUPID" or Codename == "C-PASSWORD-NOT-REQUIRED" or Codename == "C-USER-PASSWORD")
```
Comment (shainw) [inline Detections/AlsidForAD/UserAccountIssues.yaml:26]: Change to List, same as I had shown in "Alsid Active Directory attacks pathways" detection.

**Review by shainw — CHANGES_REQUESTED** [review]

**Review by v-jayakal — CHANGES_REQUESTED** [review]

File: DataConnectors/AlsidForAD.json [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
...\\" \\\"\"\r\n                     AlertID:int \"\\\" \\\"\"\r\n                     Forest:string \"\\\" \\\"\"\r\n                     Domain:string \"\\\" \"\r\n                     DistinctPart:string;\r\nlet Deviances = Common\r\n| where MessageType == 0 | parse DistinctPart with \"\\\"\"\r\n                     Codename:string \"\\\" \\\"\"\r\n                     Severity:string \"\\\" \\\"\"\r\n                     ADObject:string \"\\\" \\\"\"\r\n                     DevianceID:string \"\\\" \\\"\"\r\n                     ProfileID:string \"\\\" \\\"\"\r\n                     ReasonCodename:string \"\\\" \\\"\"\r\n                     EventID:string \"\\\"\"\r\n                     Attributes:string \"\\r\\n\";\r\nlet Changes = Common\r\n| where MessageType == 1\r\n| parse kind=regex DistinctPart with \"\\\"\"\r\n                     ADObject:string \"\\\" \\\"\"\r\n                     EventID:string \"\\\" \\\"\"\r\n                     EventType:string \"\\\" \"\r\n                     Attributes:string \"\\r?\\n\";\r\nunion Changes, Deviances\r\n| project-away DistinctPart, Product, _ResourceId, _SubscriptionId\r\n| lookup kind=leftouter CodenameTable on Codename;\r\n"
```
[the remark this answers is not in the document]
Comment (v-jayakal) [reply DataConnectors/AlsidForAD.json:17]: Can you please provide sample data in json format?

File: Parsers/Alsid/afad_parser.kql [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+"", "Not an alert",
+"C-ADM-ACC-USAGE", "Recent use of the default administrator account",
+"C-UNCONST-DELEG", "Dangerous delegation",
+"C-PASSWORD-DONT-EXPIRE", "Accounts with never expiring passwords",
+"C-USERS-CAN-JOIN-COMPUTERS", "Users allowed to join computers to the domain",
+"C-CLEARTEXT-PASSWORD", "Potential clear-text password",
+"C-PROTECTED-USERS-GROUP-UNUSED", "Protected Users group not used",
+"C-PASSWORD-POLICY", "Weak password policies are applied on users",
+"C-GPO-HARDENING", "Domain without computer-hardening GPOs",
+"C-LAPS-UNSECURE-CONFIG", "Local administrative account management"
```
Comment (v-jayakal) [inline Parsers/Alsid/afad_parser.kql:31]: missing , (comma) at the end of this line.

File: Parsers/Alsid/afad_parser.kql [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+"C-PRIV-ACCOUNTS-SPN", "Privileged accounts running Kerberos services",
+"C-REVER-PWD-GPO", "Reversible passwords in GPO",
+"C-ROOTOBJECTS-SD-CONSISTENCY", "Root objects permissions allowing DCSync-like attacks",
+"C-SDPROP-CONSISTENCY", "Ensure SDProp consistency",
+"C-SENSITIVE-CERTIFICATES-ON-USER", "Ensure SDProp consistency",
+"C-SLEEPING-ACCOUNTS", "Sleeping accounts",
+"C-USER-PASSWORD", "User account using old password",
+"C-USERS-REVER-PWDS", "Reversible passwords",
+];
+
```
Comment (v-jayakal) [inline Parsers/Alsid/afad_parser.kql:66]: Remove space on this line (line number 66), otherwise, CodenameTable will not be recognized.

**Comment by v-jayakal** [comment]
Sample data file name should be 'AlsidForADLog_CL.json' as per Sentinel Guidelines. Can you please change the name of the file from 'AlsidForAd.json'  to 'AlsidForADLog_CL.json'?

**Review by v-jayakal — CHANGES_REQUESTED** [review]

**Comment by shainw** [comment]
There is a JSON validation failure, but I am not sure it is accurate - "Incorrect Json file. File path: Sample Data/Syslog/AlsidForADLog_CL.json. Error message: Unexpected token ﻿ in JSON at position 0".  I checked the file manually and seems fine.  @sagamzu who is the author of the JSON validations, thoughts here?

**Comment by nazang** [comment]
> There is a JSON validation failure, but I am not sure it is accurate - "Incorrect Json file. File path: Sample Data/Syslog/AlsidForADLog_CL.json. Error message: Unexpected token ﻿ in JSON at position 0". I checked the file manually and seems fine. @sagamzu who is the author of the JSON validations, thoughts here?

**Comment by nazang** [comment]
> There is a JSON validation failure, but I am not sure it is accurate - "Incorrect Json file. File path: Sample Data/Syslog/AlsidForADLog_CL.json. Error message: Unexpected token ﻿ in JSON at position 0". I checked the file manually and seems fine. @sagamzu who is the author of the JSON validations, thoughts here?

Same failure for the next PR "Dark trace connector" as well, copy out the json and validate manually, seems fine. @sagamzu, could you help to take a look?

File: Workbooks/Images/Logos/Alsid.svg [hunk 9]
```diff
@@ -0,0 +1 @@
...1.1-.07-2.2-.11-3.32-.11s-2.23,0-3.33.11H363.18a5.16,5.16,0,0,0-5.17,5.17V514.3a1,1,0,0,0,1,1h13.83a1,1,0,0,0,1-1V430.05l.34.44v-.32h51.17v.14c.51,0,1,0,1.55,0a34.55,34.55,0,1,1,0,69.09c-.52,0-1,0-1.55,0v.07h-28.9a2.57,2.57,0,0,0-2.58,2.58v3a10.33,10.33,0,0,0,10.32,10.33h21.15l1.56,0,1.56,0a50.55,50.55,0,0,0,1.76-101Z"/><path class="cls-1" d="M90.49,427.78a26.9,26.9,0,0,0-46.66,0L1.48,501.11l.05,0a9.24,9.24,0,0,0,6.68,14.27,8.77,8.77,0,0,0,1,.06,9.27,9.27,0,0,0,2.84-.45c.24-.07.47-.16.7-.25l.34-.16,68-26.84a2.58,2.58,0,0,0,1.28-3.69h0a15,15,0,0,0-18.47-6.43L23.65,493.6l33.51-58a11.33,11.33,0,0,1,9.93-5.73h.15a11.33,11.33,0,0,1,9.92,5.73l46,79.69,0,.1h10.91a4.13,4.13,0,0,0,3.57-6.2L90.49,427.78Z"/><path class="cls-1" d="M308.47,487c0-18.15-11.43-26.77-38-28.92l-47.11-3.79c-18.11-1.37-22.18-6.12-22.18-12.64,0-7.53,5.61-11.19,17.16-11.19h73a10.33,10.33,0,0,0,10.33-10.33v-3.35a2.59,2.59,0,0,0-2.59-2.59H219.77c-23.72,0-36.79,9.91-36.79,27.9,0,17.08,11.94,26,36,27.92L267.79,474c15.55,1.24,22.23,5.63,22.23,13.64,0,8.39-6.64,11.55-19.21,11.55H243.14A4.12,4.12,0,0,0,239,503.3v11.47a1,1,0,0,0,1,1h31.64c23.73,0,36.79-9.91,36.79-27.9,0-.08,0-.15,0-.23S308.47,487.25,308.47,487Z"/></g></g></svg>
```
Comment (nazang) [inline Workbooks/Images/Logos/Alsid.svg:1]: The logo id attribute has to be a unique id. Could you update:

id="Layer_2"  -->  id="\<generated guid>"
id="Layer_1-2" -->  id="\<generated guid>"

**Review by nazang — CHANGES_REQUESTED** [review]

**Comment by v-jayakal** [comment]
@Azomasiel, there is a special character in AlsidForADLog_CL.json file, Please check the PR that removes that character, approve and merge into AlsidOfficial:AFAD-connector branch. OR you can do the change in your branch in AlsidForADLog_CL.json file.
PR: https://github.com/AlsidOfficial/Azure-Sentinel/pull/2/files
Logo is also invalid, it has id (which should have guids for unique identification) and also has style, which is not allowed. Please refer logo guidelines here: https://github.com/Azure/Azure-Sentinel/blob/e1665d4c4442b430d6847adc65f4b9c25f7b071f/DataConnectors/ReadMe.md.
Made changes to the logo attached to be along the guidelines and attaching it here for your verification and if you are fine with this version, please update the logo (workbook as well as data connector) with this OR any other valid logo.
[Alsid.zip](https://github.com/Azure/Azure-Sentinel/files/5682271/Alsid.zip)

**Review by v-jayakal — APPROVED** [review]

**Review by nazang — APPROVED** [review]

**Review by shainw — APPROVED** [review]

**Pull request merged**